src/app/static/src/app/apiService.ts

```diff
@@ -188,7 +188,7 @@ export class APIService<S extends string, E extends string> implements API<S, E>
     }
 
     private _handleDownloadResponse = (response: AxiosResponse) => {
-        readStream(response)
+        return readStream(response)
```

Comment:
> I think the code's all ok, but you just need to include error alerts in `DownloadResults` component to display errors for these files, as you already have for the comparison report error - I tried running this with Rob's broken hintr branch and only saw errors displayed for the comparison report.
> 
> Also a couple of other small queries:

You are right, I forgot to render error component. It has now been updated.

Reply:
> Was this change needed to actually return a result on success? So the pattern established with comparison report download action, of clearing error if response is truthy was actually not working without this change, since result was never truthy?

This is now obsolete